aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -645,6 +625,9 @@ void TensorIterator::cast_outputs() {
   for (auto& op : operands_) {
     if (op.is_output && op.original_tensor.defined() &&
         op.original_tensor.scalar_type() != op.current_dtype) {
+      if (op.original_tensor.sizes() != op.tensor.sizes()){
+        op.original_tensor.resize_as_(op.tensor).as_strided_(op.tensor.sizes(), op.tensor.strides());
+      }
```

Comment:
Is this because we are lazily resizing now? I'm surprised that `allocate_or_resize_outputs` cannot be assumed to have handled this for you.

Reply:
Yes, we used to resize before computing types, so original output was resized to correct size (but not necessarily correct layout) before intermediate output was allocated. Now, by the time we are resizing, original output is forgotten and we resize intermediate only. 
That's a good point though, I better handle this in `allocate_or_resize_outputs` anyway.